### ISSUE TITLE
Timeout errors while adding VM

### ISSUE THREAD

**Issue by n3oxmind** [GNS3/gns3-gui#2577]
GNS3 2.1.9:
I got lots of `Timeout errors`  while adding a `win10 vm` to my lab. Also, please make the cloning process run in the background because it will make the lab unusable until the cloning complete. 

**here is some ideas:**
1- I want to use the lab while the cloning in progress.
2- I think adjusting the `Timeout` base on the `base-img` size will solve most of the `Timeout errors`. 
3- get ride of the pop-up cloning window and make the progress bar shown above the `clone symbol` on the workspace.
4- I also noticed that creating the first clone take sometime whereas creating a subsequent clone will take 1-2s without even noticing it.

```Timeout after 30 seconds for request http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/computes
Error while getting compute list: Operation timeout
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/computes
Error while getting compute list: Operation timeout
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/computes
Error while getting compute list: Operation timeout
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/computes
Error while getting compute list: Operation timeout
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/computes
Error while getting compute list: Operation timeout
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Timeout after 30 seconds for request http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/computes
Error while getting compute list: Operation timeout
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/settings
Aborting request for http://riTawhZiHk02rSJIbPrR4XG1E1N69WqPRnZEnRSGjB0gdALt69Q2ZJHgajtnUWUP@127.0.0.1:3080/v2/computes
Error while getting compute list: Operation timeout```


P.S. The VM clone has been created successfully even with the `timeout errors`.

**Comment by grossmj** [comment]
> I got lots of Timeout errors while adding a win10 vm to my lab.

What kind of VM is it? Qemu, VMware or VirtualBox?

> Also, please make the cloning process run in the background because it will make the lab unusable until the cloning complete.

Just to be sure, are you talking about snapshots when you say "cloning"?

**Comment by n3oxmind** [comment]
its a Qemu VM. This is not snapshot and what i'm mean by cloning is when gns3 trying to create an overlay image base on the original (base) image (e.g dragging and dropping image from the sidebar to the workspace).

P.S. image size is 60GB format is raw

**Comment by grossmj** [comment]
> P.S. image size is 60GB format is raw

I think the raw format could be an issue. Have you tried to convert your image to qcow2 and see if this speedup things?

```
qemu-img convert -f raw -O qcow2 /path/to/your/hdd/vm01.img /path/to/your/hdd/vm01.qcow2
```

**Comment by n3oxmind** [comment]
@grossmj, I actually know why this happen. It has nothing to do with the `qemu` image format. This is happen every time gns3 tries to generate `md5sum` of an image. After generating the `md5sum` gns3 will create overlay image of the original as expected. The time is required to generate `md5sum` is not constant it depends on the image(file) size. Please, adjust the timeout in case there is no md5sum for the image that we are trying to add or say something like generating `md5sum` or separate the `md5sum` generation from the image creation function to avoid timeout error.

In short, Timeout will happen if the image you trying to use have no md5sum.

Note: to reproduce this just delete the `md5sum` of your qemu image and then try to add the image to your workplace.

Cheers!

**Comment by grossmj** [comment]
Thanks for researching the problem. It looks like the issue has been fixed on our 2.2 branch already: https://github.com/GNS3/gns3-server/pull/1275

I think we should consider to backport this into the 2.1 branch.

**Comment by grossmj** [comment]
I think it is be better to not make this change in 2.1 and risk breaking something since it is not a blocking issue. Let's close this for now and re-open if we spot this kind of problem when we release 2.2.
